Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SaveVulcanGSS.py

```diff
@@ -0,0 +1,462 @@
+                    binning_parameter_list[ws_index] = bin_parameters
+                # END-FOR
+            # END-FOR (i_row)
+        # END-IF-ELSE
+
+        # check whether there is any spectrum that does not have been listed
+        for index, bin_param in enumerate(binning_parameter_list):
+            if bin_param is None:
+                raise RuntimeError('Not all the spectra that have binning parameters set.'
+                                   '{0}-th binning parameter is None.  FYI, there are '
+                                   '{1} parameters as {2}'.format(index, len(binning_parameter_list),
+                                                                  binning_parameter_list))
+
+        return binning_parameter_list
+
+    def _process_binning_parameters(self, bin_par_str):
+        """
+        process binning parameters in string.  there are two types of binning parameters that are accepted
+        1. regular x0, dx0, x1, dx1, etc or
+        2. workspace name: workspace index
+        :param bin_par_str:
+        :return:
+        """
+        if bin_par_str.count(':') == 0:
+            # parse regular binning parameters
+            terms = bin_par_str.split(',')  # in string format
+            bin_param = list()
+            try:
+                for term in terms:
+                    bin_param.append(float(term))
+            except ValueError:
+                raise RuntimeError('Binning parameters {0} have non-float terms.'.format(bin_par_str))
+
+        elif bin_par_str.count(':') == 1:
+            # in workspace name : workspace index mode
+            terms = bin_par_str.split(':')
+            ref_ws_name = terms[0].strip()
+            if AnalysisDataService.doesExist(ref_ws_name) is False:
+                raise RuntimeError('Workspace {0} does not exist (FYI {1})'
+                                   ''.format(ref_ws_name, bin_par_str))
+            try:
+                ws_index = int(terms[1].strip())
+            except ValueError:
+                raise RuntimeError('{0} is supposed to be an integer for workspace index but not of type {1}.'
+                                   ''.format(terms[1], type(terms[1])))
+
+            ref_tof_ws = AnalysisDataService.retrieve(ref_ws_name)
+            if ws_index < 0 or ws_index >= ref_tof_ws.getNumberHistograms():
+                raise RuntimeError('Workspace index {0} must be in range [0, {1})'
+                                   ''.format(ws_index, ref_tof_ws.getNumberHistograms()))
+
+            ref_tof_vec = ref_tof_ws.readX(ws_index)
+            delta_tof_vec = ref_tof_vec[1:] - ref_tof_vec[:-1]
+
+            bin_param = numpy.empty((ref_tof_vec.size + delta_tof_vec.size), dtype=ref_tof_vec.dtype)
+            bin_param[0::2] = ref_tof_vec
+            bin_param[1::2] = delta_tof_vec
+
+            self.log().warning('Binning parameters: size = {0}\n{1}'.format(len(bin_param), bin_param))
+
+        else:
```

Comment:
Could be done in a `validateInputs()` method that will be called by the framework before `PyExec()`

Reply:
It is not straightforward to validate the inputs.  Either don't want to use the class member variables.